src/libmongoc/src/mongoc/mongoc-client-session.c

```diff
@@ -1516,12 +1569,16 @@ _mongoc_client_session_append_read_concern (const mongoc_client_session_t *cs,
       (txn_state == MONGOC_INTERNAL_TRANSACTION_STARTING || is_read_command) &&
       mongoc_session_opts_get_causal_consistency (&cs->opts) &&
       cs->operation_timestamp;
+   is_snapshot =
+      is_read_command && mongoc_session_opts_get_snapshot (&cs->opts);
```

Comment:
`is_read_command` may be true on commands outside of `find` / `aggregate` / `distinct`. E.g. `listIndexes`, or any command run through the generic "read" helpers like `mongoc_client_read_command_with_opts`.

This function may also need to accept a `cmd_name` argument, which the caller `mongoc_cmd_parts_assemble`, has access to.

Reply:
Gotcha. Used `is_find_aggregate_distinct` instead of `is_read_command` here, and added a new `cmd_name` parameter to `_mongoc_client_session_append_read_concern`.